From eccc7e7627caf33c4226d987d8ff9c7494646cf5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?I=C3=B1aki=20=C3=9Acar?= <iucar@fedoraproject.org>
Date: Sat, 30 May 2020 16:34:46 +0200
Subject: [PATCH 4/8] add support for boost 1.73

* enable BOOST_BIND_GLOBAL_PLACEHOLDERS to retain placeholders in the global namespace

* add some missing includes of <boost/bind.hpp>

(cherry picked from commit 9b9a200712cfd334ec87256f75876b91b41242a8)
---
 src/cpp/CMakeLists.txt                           | 1 +
 src/cpp/core/system/PosixChildProcessTracker.cpp | 1 +
 src/cpp/core/system/PosixOutputCapture.cpp       | 2 ++
 src/cpp/core/system/PosixSystem.cpp              | 1 +
 4 files changed, 5 insertions(+)

diff --git a/src/cpp/CMakeLists.txt b/src/cpp/CMakeLists.txt
index af791506eb..d049524dbc 100644
--- a/src/cpp/CMakeLists.txt
+++ b/src/cpp/CMakeLists.txt
@@ -28,6 +28,7 @@ include("${CMAKE_CURRENT_SOURCE_DIR}/../../CMakeGlobals.txt")
 
 # global directives
 add_definitions(-DBOOST_ENABLE_ASSERT_HANDLER)
+add_definitions(-DBOOST_BIND_GLOBAL_PLACEHOLDERS)
 
 # explicitly do not use new c++ 11 features for websocketpp
 # they currently do not work with our source
diff --git a/src/cpp/core/system/PosixChildProcessTracker.cpp b/src/cpp/core/system/PosixChildProcessTracker.cpp
index 2ee1a81dcc..976da4f6f9 100644
--- a/src/cpp/core/system/PosixChildProcessTracker.cpp
+++ b/src/cpp/core/system/PosixChildProcessTracker.cpp
@@ -18,6 +18,7 @@
 #include <sys/wait.h>
 
 #include <boost/format.hpp>
+#include <boost/bind.hpp>
 
 namespace rstudio {
 namespace core {
diff --git a/src/cpp/core/system/PosixOutputCapture.cpp b/src/cpp/core/system/PosixOutputCapture.cpp
index 762cc415e8..4c76915b82 100644
--- a/src/cpp/core/system/PosixOutputCapture.cpp
+++ b/src/cpp/core/system/PosixOutputCapture.cpp
@@ -28,6 +28,8 @@
 
 #include <core/system/System.hpp>
 
+#include <boost/bind.hpp>
+
 namespace rstudio {
 namespace core {
 namespace system {
diff --git a/src/cpp/core/system/PosixSystem.cpp b/src/cpp/core/system/PosixSystem.cpp
index c25cb32875..8dae263288 100644
--- a/src/cpp/core/system/PosixSystem.cpp
+++ b/src/cpp/core/system/PosixSystem.cpp
@@ -23,6 +23,7 @@
 
 #include <boost/algorithm/string.hpp>
 #include <boost/range/as_array.hpp>
+#include <boost/bind.hpp>
 
 #include <signal.h>
 #include <fcntl.h>
-- 
2.28.0

